submissions/ses3332002/tiny-js-world/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+const dog = {
+  species: "dog",
+  name: "Toby",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Woof-woof!",
+  friends: [
+    "Puss"
+  ]
+};
+
+const cat = {
+  species: "cat",
+  name: "Puss",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Meow!",
+  friends: [
+    "Toby",
+    "Halley"
+  ]
+};
+
+const man = {
+  species: "human",
+  name: "Jack",
+  gender: "male",
+  legs: 2,
+  hands: 2,
+  saying: "Hey baby!",
+  friends: [
+    "Jennie",
+    "Halley"
+  ]
+};
+
+const woman = {
+  species: "human",
+  name: "Jennie",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: "Hi darling!",
+  friends: [
+    "Jack",
+    "Toby"
+  ]
+};
+
+const catWoman = {
+  species: "super-hero",
+  name: "Halley",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: cat.saying,
+  friends: [
+    "Puss"
+  ]
+};
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+
+inhabitants.forEach((item) => {
+  let description = "";
+  for (let key in item) {
+    if (typeof(item[key]) == "object") {
```

Comment:
Perhaps `Array.isArray` would work better here.

Reply:
:) Well, may be it works better, but the result is exactly the same.
I will remake it with Array.isArray.

Reply:
Think of future extension. If someone adds a property like `this.wings = { span: 30, canfly: true };` this code will try to apply `.join` and throw an exception.

Whereas there is an appropriate and more precise tool, just use it to have less headache maintaining codebase. Take care of yourself future if not of your peer developers.

Reply:
Both `console.log(0); console.log(1); console.log(2); console.log(3); ...` and `for (let i = 0; i < 3; i++) console.log(i);` also give exactly the same result ;-)

Reply:
I remade array checking with Array.isArray.